test/components/tree-view.spec.tsx

```diff
@@ -128,314 +116,376 @@ describe('<TreeView />', () => {
+                    treeView.pressKey(keycode('left'));
 
-                    return waitForDom(() => expect(
-                        isElementFocused(select(getTreeItem(treeData[0].label))!, treeNodeStyle)).to.equal(true));
+                    await waitForDom(() => expect(
+                        isElementFocused(treeView.getItem(treeData[0].label)!, treeNodeStyle)).to.equal(true));
                 });
 
             it('moves to child to if there is one after expanding the element if possible when right is clicked',
                 async () => {
-                    const {select, waitForDom} = clientRenderer.render(<TreeViewDemo />);
+                    const {driver, waitForDom} = clientRenderer.render(
+                        <TreeViewDemo />
+                    ).withDriver(TreeViewDemoDriver);
+
+                    const {treeView} = driver;
 
                     const nodeChildren = treeData[0].children;
 
-                    expandItemWithLabel(select, treeData[0].label);
+                    treeView.toggleItem(treeData[0].label);
 
-                    await waitForDom(() => expect(select(getTreeItem(nodeChildren![0].label))).to.be.present());
+                    await waitForDom(() => expect(treeView.getItem(nodeChildren![0].label)).to.be.present());
 
                     await waitForDom(() => expect(
-                        isElementFocused(select(getTreeItem(treeData[0].label))!, treeNodeStyle)).to.equal(true));
+                        isElementFocused(treeView.getItem(treeData[0].label)!, treeNodeStyle)).to.equal(true));
 
-                    simulate.keyDown(select('TREE_VIEW_DEMO', 'TREE_VIEW'), {keyCode: KeyCodes.RIGHT});
+                    treeView.pressKey(keycode('right'));
 
-                    return waitForDom(() => expect(
-                        isElementFocused(select(getTreeItem(nodeChildren![0].label))!, treeNodeStyle)).to.equal(true));
+                    await waitForDom(() => expect(
+                        isElementFocused(treeView.getItem(nodeChildren![0].label)!, treeNodeStyle)).to.equal(true));
                 });
 
             it('focuses next and previous when down and up arrows are clicked', async () => {
-                const {select, waitForDom} = clientRenderer.render(<TreeViewDemo />);
+                const {driver, waitForDom} = clientRenderer.render(
+                    <TreeViewDemo />
+                ).withDriver(TreeViewDemoDriver);
 
-                const rootNode = getTreeItem(treeData[0].label);
+                const {treeView} = driver;
+
+                const rootNode = treeView.getItem(treeData[0].label);
                 const nodeChildren = treeData[0].children;
 
-                selectItemWithLabel(select, treeData[0].label);
-                expandItemWithLabel(select, treeData[0].label);
+                treeView.selectItem(treeData[0].label);
+                treeView.toggleItem(treeData[0].label);
```

Comment:
lets have a `toggle(): void` method in `TreeItemDriver`

Reply:
I think its OK having it on the TV driver, but it should invoke the getters of the item driver in the implementation